compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -1039,10 +1040,15 @@ DiagnosedSilenceableFailure transform_dialect::ConfigExtractPart::apply(
     transformResults.set(getResultConfigPart().cast<OpResult>(), {});
     return emitSilenceableFailure(target) << " no tiling at level";
   }
+  SmallVector<Value> values;
   SmallVector<Operation *> results;
   OpBuilder b(target);
-  for (int64_t ts : vals)
+  for (int64_t ts : vals) {
     results.push_back(b.create<arith::ConstantIndexOp>(target->getLoc(), ts));
+    values.push_back(results.back()->getResult(0));
+  }
+  b.create<LinalgExt::DoNotDCEOperandsOp>(target->getLoc(), values);
```

Comment:
I'm wondering if this is the right way to fix the DCE issue. Wouldn't it be better if we could query the transform interpreter if an op is a payload mapped to some handle. If so, we exclude it from DCE (and potentially other "cleanup" transforms).

Reply:
We don't want to modify canonicalization and CSE with custom one-off things IMO.
We already keep our own Listener-version copy of those that is painful to maintain.